CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -1415,9 +1433,23 @@ CFTypeRef _Nullable _CFThreadSpecificGet(_CFThreadSpecificKey key) {
 void _CFThreadSpecificSet(_CFThreadSpecificKey key, CFTypeRef _Nullable value) {
 #if TARGET_OS_WIN32
     if (value != NULL) {
+        _CFThreadSpecificData *oldData = (_CFThreadSpecificData *)FlsGetValue(key);
+        if (oldData) {
+            free(oldData);
```

Comment:
Probably should add a comment explaining that the value should not be freed here.